lib/ProMotion/table/table.rb

```diff
@@ -299,7 +299,13 @@ def tableView(_, viewForHeaderInSection: index)
     def tableView(_, heightForHeaderInSection: index)
       section = promotion_table_data.section(index)
       if section[:title_view] || section[:title].to_s.length > 0
-        section[:title_view_height] || tableView.sectionHeaderHeight
+        if section[:title_view_height]
+          section[:title_view_height]
+        elsif tableView(_, viewForHeaderInSection: index).respond_to?(:height)
+          tableView(_, viewForHeaderInSection: index).height
```

Comment:
@jamonholmgren is there a way we can optimize this so we're not calling the method twice? maybe something like:

```ruby
elsif header_class = tableView(_, viewForHeaderInSection: index) && header_class.respond_to?(:height)
  header_class.height
end
```

I've never been good at optimizing stuff like this.